activerecord/lib/active_record/connection_adapters/abstract/transaction.rb

```diff
@@ -90,6 +98,11 @@ def commit_records
         ite.each do |i|
           i.committed!(should_run_callbacks: false)
         end
+        @records.uniq.each do |i|
+          if i.weakref_alive?
+            i.committed!(should_run_callbacks: false)
+          end
```

Comment:
Does this mean that only objects which *haven't* been GC'd will get the `committed!` callback?